src/main/java/hudson/remoting/RemoteClassLoader.java

```diff
@@ -839,19 +839,20 @@ public ClassFile fetch2(String className) throws ClassNotFoundException {
         /**
          * Fetch a single class and creates a {@link ClassFile2} for it.
          */
-        public ClassFile2 fetch4(String className, ClassFile2 referer) throws ClassNotFoundException {
+        public ClassFile2 fetch4(String className, @CheckForNull ClassFile2 referer) throws ClassNotFoundException {
+            Class<?> referrerClass = referer == null ? null : referer.clazz;
```

Comment:
Should this case not be logged? I have no idea if this is common or not.
Maybe as FINEST or so?

Reply:
That `referer == null`? No, that is the common case actually, according to `fetch3`: starts with it null, then sets `referer` if it is doing bytecode analysis. The naming is not very informative, admittedly.